examples/platform/nrfconnect/pw_sys_io/sys_io_nrfconnect.cc

```diff
@@ -0,0 +1,63 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <cinttypes>
+
+#include "pw_sys_io/sys_io.h"
+#include <zephyr.h>
+#include <assert.h>
+#include "console/console.h"
+
+extern "C" void pw_sys_io_Init()
+{
+    assert(console_init() == 0);
+}
+
+namespace pw::sys_io {
+
+Status ReadByte(std::byte * dest)
+{
+    assert(dest);
+    const int c = console_getchar();
+    *dest = static_cast<std::byte>(c);
```

Comment:
You could use SafeInt.h
if (!chip::CanCastTo<std::byte>(c))
  return Status:::INTERNAL;
...